src/TiledArray/dist_array.h

```diff
@@ -56,12 +56,22 @@ template <typename Tile = Tensor<double, Eigen::aligned_allocator<double>>,
 class DistArray : public madness::archive::ParallelSerializableObject {
  public:
   typedef DistArray<Tile, Policy> DistArray_;  ///< This object's type
-  typedef TiledArray::detail::ArrayImpl<Tile, Policy> impl_type;
+  typedef TiledArray::detail::ArrayImpl<Tile, Policy>
+      impl_type; ///< The type of the PIMPL
   typedef typename impl_type::policy_type policy_type;  ///< Policy type
-  typedef typename detail::numeric_type<Tile>::type
-      element_type;  ///< The tile element type
-  typedef typename detail::scalar_type<Tile>::type
-      scalar_type;  ///< The tile scalar type
+
+  /// Type used to hold the components of the tensors in the array. For
+  /// DistArray<Tensor<double>> this will be double. Similarly for
+  /// DistArray<Tensor<Tensor<double>> this will also be double. Notably for
+  /// complex elements of type std::complex<T> element_type will be
+  /// std::complex<T> and scalar_type will be T.
+  typedef typename detail::numeric_type<Tile>::type element_type;
```

Comment:
perhaps better name is `numeric_type`?

Reply:
We refer to "elements" of the DistArray as element in `elements_range` etc. Elements of `DistArray<Tensor<Tensor<T>>>` are indeed `Tensor<T>`, so it makes sense to avoid calling `T` by `element_type`